io.openems.backend.metadata.user_based/src/io/openems/backend/metadata/user_based/UserBased.java

```diff
@@ -0,0 +1,207 @@
+        return Collections.unmodifiableCollection(this.edges.values());
+    }
+
+    /**
+     * This method checks whether the requested channels are available for the user with the given user id
+     * @param userId the requesting user
+     * @param edgeId the id of the edge of the channels
+     * @param requestedChannels the requested channels
+     * @return all permitted channels
+     * @throws OpenemsException gets thrown in case the user is not permitted of accessing the edge
+     */
+    @Override
+    public TreeSet<ChannelAddress> checkChannels(String userId, String edgeId, TreeSet<ChannelAddress> requestedChannels) throws OpenemsException {
+        if (!this.getUser(userId).isPresent()) {
+            throw new OpenemsException("There is no user with id (" + userId + ")");
+        }
+        BackendUser user = checkPermissionAndThrow(userId, edgeId);
+
+        // the user also has the permission to access the edge
+        List<ChannelAddress> permittedChannels = user.getChannels(edgeId);
+        if (permittedChannels == null || permittedChannels.isEmpty()) {
+            // Seems like there is an edge configured without any channels which may get accessed
+            return new TreeSet<>();
+        }
+
+        TreeSet<ChannelAddress> retVal = new TreeSet<>(permittedChannels);
+        retVal.retainAll(requestedChannels);
+        return retVal;
+    }
+
+    /**
+     * Checks whether the user is permitted to access the given edge
+     * @param userId id of user
+     * @param edgeId if of edge
+     * @return the requested user
+     * @throws OpenemsException gets thrown if the permission is missing
+     */
+    private BackendUser checkPermissionAndThrow(String userId, String edgeId) throws OpenemsException {
+        // there is a user configured for the given userId
+        BackendUser user = this.getUser(userId).get();
+        if (user.getEdgeRoles().get(edgeId) != null && !user.getEdgeRoles().get(edgeId).isAtLeast(Role.GUEST)) {
+            throw new OpenemsException("User (" + userId + ") does not have the permission to access the edge (" + edgeId + ")");
+        }
+        return user;
+    }
+
+
+    /**
+     * In case there is a path for a configuration file configured, this method extracts the JSON-encoded information
+     * and fills the fields of the class.<br>
+     * See also: {@link UserBased#userMap}, {@link UserBased#edges}
+     */
+    private synchronized void refreshData() {
+        if (!this.edges.isEmpty()) {
+            return;
+        }
+
+        StringBuilder sb = FileUtils.checkAndGetFileContent(this.path);
+        if (sb == null) {
+            // exception occurred. File could not be read
+            return;
+        }
+
+        List<Edge> edges = new ArrayList<>();
+
+        // parse to JSON
+        try {
+            JsonElement config = JsonUtils.parse(sb.toString());
+            JsonArray jUsers = JsonUtils.getAsJsonArray(config, JsonKeys.USERS.value());
```

Comment:
In my opinion, this JsonKeys enum is over-engineered and makes the code harder to read. If I were you I would just use plain string identifiers here. I do not quite see the benefit of making the Enum reusable within `io.openems.common.utils`